### PR TITLE
pep8 compliance

### PR DESCRIPTION
**Pull request by ComputerGenius152**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,9 +12,9 @@ If you have logged in with an administrator account, you will be able to manage
 Contributors are welcome. If you want to help, feel free to submit a pull request or open an issue.
 
 #### Code Style
-4 space tabs. No tab characters, no 2 space tabs, no 8 space tabs.<br>
-Max line length is 120 characters, but preferably keep it below 100.<br>
-Make it readable. Use descriptive variable names, and define them close to the first use.
+Follow the [PEP 8](https://www.python.org/dev/peps/pep-0008/) style guide for Python code.
+
+For non-Python code, please use 4 space tabs, limit your lines to under 120 characters, and use descriptive variable names.
```
Comment (jdabtieu) [inline README.md:17]: Change to under 100 characters

File: application.py [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
 from cs50 import SQL
-from flask import Flask, redirect, render_template, request, session, send_from_directory, abort, url_for, flash
+from flask import (Flask, abort, flash, redirect, render_template, request,
+                   send_from_directory, session, url_for)
+from flask_mail import Mail, Message
 from flask_session import Session
-from tempfile import mkdtemp
-from werkzeug.exceptions import default_exceptions, HTTPException, InternalServerError
+from helpers import (admin_required, contest_retrieve, generate_password,
+                     login_required, send_email)
```
Comment (jdabtieu) [inline application.py:12]: These should be in a third group by themselves, as they are local imports

File: application.py [hunk 3]
```diff
@@ -222,19 +248,25 @@ def forgotpassword():
...
+    rows = db.execute("SELECT * FROM users WHERE email = :email",
+                      email=request.form.get("email"))
 
     if len(rows) == 1:
         exp = datetime.now() + timedelta(seconds=600)
-        token = jwt.encode({'user_id': rows[0]["id"], 'expiration': exp.isoformat()}, app.config['SECRET_KEY'], algorithm='HS256').decode('utf-8')
-        text = render_template('email/reset_password_text.txt', username=rows[0]["username"], token=token)
-        send_email('Reset Your CTF Password', app.config['MAIL_DEFAULT_SENDER'], [email], text, mail)
+        token = jwt.encode({'user_id': rows[0]["id"], 'expiration': exp.isoformat(
+        )}, app.config['SECRET_KEY'], algorithm='HS256').decode('utf-8')
```
Comment (jdabtieu) [inline application.py:257]: oh no no you can't split the paranthesis like that
split it at the comma

File: application.py [hunk 4]
```diff
@@ -510,18 +572,23 @@ def editproblem(problem_id):
...
 
     new_name = request.form.get("name")
-    new_description = request.form.get("description").replace("\r", "").replace("\n", "<br>\n")
+    new_description = request.form.get("description").replace(
+        "\r", "").replace("\n", "<br>\n")
     new_hint = request.form.get("hints")
     if new_hint:
-        new_hint = request.form.get("hints").replace("\r", "").replace("\n", "<br>\n")
+        new_hint = request.form.get("hints").replace(
+            "\r", "").replace("\n", "<br>\n")
```
Comment (jdabtieu) [inline application.py:580]: this just makes the code less readable

File: application.py [hunk 5]
```diff
@@ -535,14 +602,18 @@ def problem_editeditorial(problem_id):
 
     # Reached via POST
 
-    new_editorial = request.form.get("editorial").replace("\r", "").replace("\n", "<br>\n")
+    new_editorial = request.form.get("editorial").replace(
+        "\r", "").replace("\n", "<br>\n")
```
Comment (jdabtieu) [inline application.py:606]: see above comment

File: application.py [hunk 6]
```diff
@@ -565,22 +636,28 @@ def admin_submissions():
 
     if request.args.get("correct"):
         modifier += " correct=? AND"
-        args.insert(len(args), True if request.args.get("correct") == "AC" else False)
+        args.insert(len(args), True if request.args.get(
+            "correct") == "AC" else False)
```
Comment (jdabtieu) [inline application.py:640]: change to
`args.insert(len(args), request.args.get("correct") == "AC")`

File: application.py [hunk 7]
```diff
@@ -616,18 +695,19 @@ def admin_createcontest():
     if check_end < check_start:
         return render_template("admin/createcontest.html", message="Contest cannot end before it starts!"), 400
 
-
-    description = request.form.get("description").replace("\r", "").replace("\n", "<br>\n")
+    description = request.form.get("description").replace(
+        "\r", "").replace("\n", "<br>\n")
```
Comment (jdabtieu) [inline application.py:699]: once again, this makes code less readable

File: application.py [hunk 8]
```diff
@@ -641,14 +721,16 @@ def createproblem():
 
     problem_id = request.form.get("id")
     name = request.form.get("name")
-    description = request.form.get("description").replace("\r", "").replace("\n", "<br>\n")
+    description = request.form.get("description").replace(
+        "\r", "").replace("\n", "<br>\n")
```
Comment (jdabtieu) [inline application.py:725]: see above comment

File: application.py [hunk 9]
```diff
@@ -661,20 +743,23 @@ def createproblem():
...
 @app.route("/admin/ban")
 @admin_required
 def ban():
     user_id = request.args.get("user_id")
     if not user_id:
         return "Must provide user ID"
 
-    banned_status = db.execute("SELECT banned FROM users WHERE id=:id", id=user_id)
+    banned_status = db.execute(
+        "SELECT banned FROM users WHERE id=:id", id=user_id)
```
Comment (jdabtieu) [inline application.py:762]: i'm a bit confused here,

on one line you split the query into two lines, first one contains the string, second one contains the parameter

then on the second instance, you just moved both the query and parameter to a second line

any reason for this?

File: application.py [hunk 10]
```diff
@@ -715,13 +804,16 @@ def createannouncement():
         return render_template("admin/createannouncement.html", message="You have not entered all required fields"), 400
 
     name = request.form.get("name")
-    description = request.form.get("description").replace("\r", "").replace("\n", "<br>\n")
+    description = request.form.get("description").replace(
+        "\r", "").replace("\n", "<br>\n")
```
Comment (jdabtieu) [inline application.py:808]: this is less readable

File: application.py [hunk 11]
```diff
@@ -738,16 +830,19 @@ def editannouncement(a_id):
 
     # Reached via POST
     new_name = request.form.get("name")
-    new_description = request.form.get("description").replace("\r", "").replace("\n", "<br>\n")
+    new_description = request.form.get("description").replace(
+        "\r", "").replace("\n", "<br>\n")
```
Comment (jdabtieu) [inline application.py:834]: see above

File: application.py [hunk 12]
```diff
@@ -764,7 +859,8 @@ def editcontest(contest_id):
...
```
Comment (jdabtieu) [inline application.py:863]: see above

**Review by jdabtieu — CHANGES_REQUESTED** [review]
Did you use an automated tool to do this? There's inconsistent SQL queries
```python
    rows = db.execute("SELECT * FROM users WHERE id = :id",
                      id=session["user_id"])
```
```python
    rows = db.execute(
        "SELECT * FROM users WHERE email = :email", email=token["email"])
```
(both of the above snippets can be written in either way without issue)

and then a bunch of lines surpass 79 characters easily
```python
        return render_template("contest/contest_problem.html", data=check[0], status="fail", message="Your flag is incorrect.")
```

File: README.md (hunk 1 above)
Comment (jdabtieu) [reply README.md:17]: I also want to change the Python line limit to 90. I know PEP 8 specifies 79 but that's frankly annoying sometimes (and they say some people do make it longer, up to a maximum of 99). Even with 90, it's short enough that if someone were to split screen, code still wouldn't spill past the edge of the window

**Comment by ComputerGenius152** [comment]
Huh, I really need to stop using autopep8.

**Comment by ComputerGenius152** [comment]
It should be more consistent now, and I've edited the code style to allow for longer lines that increase readability.

**Pull request merged**